odoo/index.py

```diff
@@ -214,6 +217,10 @@ class unique(index, _unique):
     message: str = ''
     method = 'btree'
 
+    def expression(self, env, field):
+        "A unique index is useful even on a translated field"
```

Comment:
really, is it working with the JSON conversion?

Reply:
IMO, unique + translated field doesn't work as expected. It compares the entire JSON, but doesn't take into account the fallback language. Also, developers expect uniqueness for each language, and not for the 'sum' of them.
We should warm this case (or make it work as expected, but I cannot imagine the cost of that)